### ISSUE TITLE
Tagging words containing hyphen does not work

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#977]
Found another issue with hyphens...

```
const nlp = require('compromise')
nlp.verbose('tagger')

nlp.plugin({
  words: {
    'x-ray': 'Diagnostic',
  },
})

nlp(`x-ray`).debug()
```

The term `x-ray` seems to get tagged as 2 separate terms, and so never gets the `#Diagnostic` tag:

```ascii
 ray         → #Noun                    1-lexicon
 x           → #Noun                    3-[fallback]
 x           → #Singular                3-singular-guess
 ray         → #Singular                3-singular-guess
        ↓ - 'x' (#Noun)  
 ray         → #Singular                3-[switch] (Person|Noun)
 ray         → #Noun                     -inferred by #Singular

  ┌─────────
  │ 'x'        - Noun, Singular
  │ 'ray'      - Noun, Singular
```

**Comment by thegoatherder** [comment]
@spencermountain - thanks for this fix.

I was interested to see the fix is to define the word `x-ray` in [data/lexicon/nouns/singulars.js](https://github.com/spencermountain/compromise/commit/041c92a3cfb1aad9a9d86ddbddc8dd418997e67d#diff-25d7ffe8c772a9244dbf6021f8807597467da083dd750b6a692a6c8a9ba99fa4)

Does this imply that all words containing hyphen are seen as two-words in Compromise unless they are hard-coded into the `singulars` lexicon?

What would be the plugin syntax for this update if we wanted to fix via plugin? (for future ref) I tried the following but got the same output as reported above:

```javascript
const nlp = require('compromise')
nlp.verbose('tagger')

nlp.plugin({
  words: {
    'x-ray': 'Diagnostic',
  },
  tags: {
    'x ray': { isA: 'Singular' },   // <-- Is this the correct fix? Does not seem to work
  },
})

nlp(`x-ray`).debug()
```

Outputs:

```ascii
ray         → #Noun                    1-lexicon
 x           → #Noun                    3-[fallback]
 x           → #Singular                3-singular-guess
 ray         → #Singular                3-singular-guess
        ↓ - 'x' (#Noun)  
 ray         → #Singular                3-[switch] (Person|Noun)
 ray         → #Noun                     -inferred by #Singular

  ┌─────────
  │ 'x'        - Noun, Singular
  │ 'ray'      - Noun, Singular
```

**Comment by spencermountain** [comment]
fixed in 14.7.0
cheers

**Comment by thegoatherder** [comment]
Thank you!
